### PR TITLE
Add rbd fio test to exercise block IO on all format/features

### PR DESCRIPTION
**Pull request by vasukulkarni**
This test can be used to exercise all rbd supported formats/features and can be used along with thrash/powercuts,  Can be used with large disks and simulate VM installation test cases. I am still testing this with other thrash tests for complete results.

Signed-off-by: Vasu Kulkarni <vasu@redhat.com>

### CODE REVIEW

**Comment by vasukulkarni** [comment]
@jdurgin @dillaman: can you review and let me know what format/features to add for config
1st yaml run log:  http://paste.ubuntu.com/12240259/
2nd task with thrash: http://paste.ubuntu.com/12240236/

**Comment by vasukulkarni** [comment]
@dillaman : fio doesn't have a good binaries for all distro's so thats the reason I am using the source, for the librbd enginet, I think I only need librbd-devel to be installed for it to work and probably has to be compiled from source,  I will try to get a good run with librbd engine and update here.

**Comment by vasukulkarni** [comment]
@dillaman @jdurgin: Added test for librbd using rbd engine (format 2, features: 1,2,4) 
Logs: http://paste.ubuntu.com/12267641/

**Comment by vasukulkarni** [comment]
@jdurgin @dillaman I am planning to merge this after I get a good run, Let me know if you need any more changes, it now tests on librbd clones using test-clone-io option, right now there are failures which seem unrelated to scripts here http://pulpito.ceph.com/ubuntu-2015-09-30_17:21:12-rbd-infernalis---basic-multi/ and I have also scheduled a krbd run.

File: tasks/rbd_fio.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+           remote.run(args=['sudo', 'rbd', 'info', 'i{i}f{f}{sn}'.format(i=frmt,f=feature,sn=sn)])
+           if ioengine != 'rbd':
+               out=StringIO.StringIO()
+               remote.run(args=['sudo', 'rbd', 'map', 'i{i}f{f}{sn}'.format(i=frmt,f=feature,sn=sn)],stdout=out)
+               dev=re.search(r'(/dev/rbd\d+)',out.getvalue())
+               rbd_dev=dev.group(1)
+               if config.get('test-clone-io'):
+                    log.info("Testing clones using fio")
+                    remote.run(args=['sudo', 'rbd', 'snap', 'create', 'i{i}f{f}{sn}@i{i}f{f}{sn}Snap'.format(i=frmt,f=feature,sn=sn)])
+                    remote.run(args=['sudo', 'rbd', 'snap', 'protect', 'i{i}f{f}{sn}@i{i}f{f}{sn}Snap'.format(i=frmt,f=feature,sn=sn)])
```
Comment (zmc) [inline tasks/rbd_fio.py]: I notice repeating the same string templates many times. Wouldn't it be nicer to only do that once?

File: tasks/rbd_fio.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+from teuthology.parallel import parallel
+from teuthology import misc as teuthology
+from tempfile import NamedTemporaryFile
+from teuthology.orchestra import run
+
+log = logging.getLogger(__name__)
+
+@contextlib.contextmanager
+def task(ctx, config):
```
Comment (zmc) [inline tasks/rbd_fio.py]: Curious, why not use ``teuthology.task.Task`` ?

File: tasks/rbd_fio.py (hunk 1 above)
Comment (vasukulkarni) [inline tasks/rbd_fio.py]: yes agree with the repetition, will fix that.

File: tasks/rbd_fio.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                     p.spawn(run_fio, remote, config[client_config])
+            
+    yield
+    
+    
+def run_fio(remote, config):
+    """
+    create fio config file with options based on above config
+    get the fio from github, generate binary, and use it to run on
+    the generated fio config file
```
Comment (zmc) [inline tasks/rbd_fio.py]: If it were me, for easier readability and maintenance, I would want to split this function up into the logical sections outline in this docstring.

File: tasks/rbd_fio.py (hunk 2 above)
Comment (vasukulkarni) [inline tasks/rbd_fio.py]: yeah sorry, I always wanted to use the new Task style but just slipped, I will start using the new Task style going forward.

**Comment by zmc** [comment]
hmm. merging your own PR would seem to defeat the purpose of review...

**Comment by vasukulkarni** [comment]
@zmc feel free to undo this and let me know if you have more comments to fix, I had some discussion with @jdurgin before and I have been testing this for quite some time.

**Comment by zmc** [comment]
@vasukulkarni I don't see any other feedback here aside from mine and I didn't consider it resolved.

If everyone else is happy with it, fine. Unlike in ``teuthology``, I don't really end up maintaining this code.

**Comment by vasukulkarni** [comment]
yeah sorry my bad, I think I should have mentioned @jdurgin is OK with this test and I added clone on io based on his feedback.

**Comment by jdurgin** [comment]
for the record I said I was ok with the rbd coverage prior to zack's review, I didn't look at the task itself too closely, and didn't expect this to be merged without resolving zack's comments

**Pull request merged**